### PR TITLE
Support TLS for MySQL

### PR DESCRIPTION
**Pull request by mrgnr**
This PR makes it possible for clients to connect to MySQL using TLS. `mysql_client_key` and `mysql_client_cert` must be set to the contents of the client's key and certificate. `mysql_ca_cert` must be set to the contents of the CA certificate used to sign the client certificate. For Google Cloud SQL, certificates can be generated using [`ax create-cloud-sql-certs`](https://github.com/appsembler/ax/pull/8).

Each service that uses a TLS connection must set the corresponding `*_MYSQL_OPTIONS` variable. For example:

```
mysql_options:
    ssl:
        key: '{{ mysql_client_key_path }}'
        cert: '{{ mysql_client_cert_path }}'
        ca: '{{ mysql_ca_cert_path }}'

ANALYTICS_API_DEFAULT_MYSQL_OPTIONS: "{{ mysql_options }}"
ANALYTICS_API_REPORTS_MYSQL_OPTIONS: "{{ mysql_options }}"
COURSE_DISCOVERY_MYSQL_OPTIONS: "{{ mysql_options }}"
ECOMMERCE_MYSQL_OPTIONS: "{{ mysql_options }}"
EDX_NOTES_API_MYSQL_OPTIONS: "{{ mysql_options }}"
EDXAPP_MYSQL_OPTIONS: "{{ mysql_options }}"
INSIGHTS_MYSQL_OPTIONS: "{{ mysql_options }}"
PROGRAMS_MYSQL_OPTIONS: "{{ mysql_options }}"
XQUEUE_MYSQL_OPTIONS: "{{ mysql_options }}"
```

### CODE REVIEW

**Comment by mrgnr** [comment]
[(6) Write Ansible role to configure TLS certificates for Google Cloud SQL](https://trello.com/c/azmSwmAh/717-6-write-ansible-role-to-configure-tls-certificates-for-google-cloud-sql)

**Comment by tkeemon** [comment]
@mrgnr this looks good to me, but I'm not the best person to be reviewing Ansible code. Is there any way for us to push this upstream or is it too tied to our own use of MySQL?

**Comment by mrgnr** [comment]
@tkeemon Yeah, I'd like to get the ability to set `OPTIONS` pushed upstream. `mysql_init` is our own custom role though, so we don't need to push that.

File: playbooks/roles/ansible-role-django-ida/templates/defaults/main.yml.j2 [hunk 1]
```diff
@@ -29,6 +29,7 @@
     PORT: '3306'
     ATOMIC_REQUESTS: true
     CONN_MAX_AGE: 60
+    OPTIONS: '{{ '{{' }} {{ role_name|upper}}_MYSQL_OPTIONS }}'
```
Comment (johnbaldwin) [inline playbooks/roles/ansible-role-django-ida/templates/defaults/main.yml.j2]: I guess from looking at this that the inner '{{' doesn't need to be escaped? I'm not clear on what purpose it serves

**Comment by johnbaldwin** [comment]
@mrgnr I just have one comment on a string with an inner '{{' which I'm not clear on. Otherwise looks straightforward to me

File: playbooks/roles/ansible-role-django-ida/templates/defaults/main.yml.j2 (hunk 1 above)
Comment (mrgnr) [inline playbooks/roles/ansible-role-django-ida/templates/defaults/main.yml.j2]: @johnbaldwin This is edX's meta-role to generate Ansible code for other IDA roles. This is crazy Jinja escape syntax.. the inner `'{{'` is the thing that gets printed and the surrounding `{{ }}` are the escapers. The syntax in this line is consistent with the syntax elsewhere in this file, so you can trust that it works :)

Comment (johnbaldwin) [inline playbooks/roles/ansible-role-django-ida/templates/defaults/main.yml.j2]: @mrgnr Thanks for the explanation!

**Pull request merged**